src/library/compiler/inliner.cpp

```diff
@@ -129,11 +139,71 @@ class inline_simple_definitions_fn : public compiler_step_visitor {
         return ctx().reduce_projection(e);
     }
 
-    virtual expr visit_app(expr const & e) override {
+    optional<expr> apply_overrides_rec(expr const & e) {
+        if (is_constant(e)) {
+            name n = const_name(e);
+            if (auto override_name = get_vm_override_name(m_env, n)) {
+                n = override_name.value();
+                while (auto override_name = get_vm_override_name(m_env, n)) {
+                    n = override_name.value();
+                }
+                return optional<expr>(mk_constant(n, const_levels(e)));
+            }
+        } else if (is_app(e)) {
+            auto fn = app_fn(e);
+            auto arg = app_arg(e);
+            bool changed = false;
+            if (auto fn2 = apply_overrides_rec(fn)) {
+                fn = *fn2;
+                changed = true;
+            }
+            if (auto arg2 = apply_overrides_rec(arg)) {
+                arg = *arg2;
+                changed = true;
+            }
+            if (changed) {
+                return optional<expr>(mk_app(fn, arg));
+            }
+        }
+        return optional<expr>();
+    }
+
+    expr apply_overrides(expr const & e) {
+        if (m_enable_overrides) {
```

Comment:
Since overrides are already applied in inlining, this means that disabling the `vm.override` option doesn't have effect on already compiled definitions that are not inlined.

I'm happy with this, but this should be mentioned somewhere.

Reply:
Actually, it has an effect because the vm uses `get_vm_decl` to resolve non-inlined calls and `get_vm_decl` takes overrides into account.

Reply:
Yes, non-inlined calls take the `vm.override` option into account.  Note: many calls are inlined.  For example all calls to function defined using the equation compiler are inlined:
```lean
set_option trace.compiler.optimize_bytecode true
-- list.length is never called
#eval [1,2,3].length
```

Again, this shouldn't be an issue if you want to use it for the qpf implementation but it should be documented.  And it makes me wonder if the `vm.override` option is actually useful in practice.

Reply:
To be honest, I'm not sold on that flag either but @digama0 seemed to think it was important. It is hard to implement in a consistent way.